enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,446 @@
+---
+title: ingress-node-firewall-with-ebpf
+
+authors:
+  - "@msherif1234"
+
+reviewers:
+  - "@danwinship"
+  - "@dave-tucker"
+  - "@dcbw"
+  - "@trozet"
+
+approvers:
+  - "@danwinship"
+  - "@knobunc"
+
+api-approvers:
+  - "@danwinship"
+
+creation-date: 2022-06-15
+
+last-updated: 2022-06-21
+
+status: implementable
+
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2781
+---
+
+# Ingress node Firewall using eBPF
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OCP customers would like to secure OCP's nodes from external attacks by configuring ingress firewall policies.
+eBPF provides a very flexible filtering mechansim to allow early detection and filtering.
+
+New cluster scoped object `IngressNodeFirewall` will be used to configure node ingress firewall match rules and a per matching rule statistics will be avaliable.
```

Comment:
Have customers asked for, or do we anticipate them asking for rule logging also?

Reply:
in the form of emitting events for packet drops ? or u mean something else ?

Reply:
yeah, basically [this](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/configuring_logging_for_denied_packets) but for your feature.